.github/workflows/upgrade.yml

```diff
@@ -85,55 +81,32 @@ jobs:
           echo "Checked out source code at $GIT_TAG."
 
       - name: Prepare the fleet using the before upgrade version
-        run: cd test/upgrade && chmod +x setup.sh && ./setup.sh 3 && cd -
+        # Set up the Fleet using images built the older source code but with the current setup script.
+        run: |
+          git checkout $PREVIOUS_COMMIT -- test/upgrade
```

Comment:
Why are we checking out to the $PREVIOUS_COMMIT, it holds the latest branch's commit,

From previous step,
PREVIOUS_BRANCH=$(git branch --show-current)
PREVIOUS_COMMIT=$(git rev-parse HEAD)

nit: the var name is confusing

Reply:
Oh, `git checkout COMMIT -- dir` will check out the current version of the code for the specific dir (without touching other parts of the directory -> this is needed so that we could always use the scripts from now instead of from before). Let me tweak the var name and add a comment to be more specific.